WordPress/Classes/Models/Notifications/NotificationSettings.swift

```diff
@@ -0,0 +1,185 @@
+            case Device         = "device"
+            
+            
+            /**
+            *  @details Returns the localized description of the current enum value
+            */
+            func description() -> String {
+                switch self {
+                case .Timeline:
+                    return NSLocalizedString("Notifications Tab", comment: "WordPress.com Notifications Timeline")
+                case .Email:
+                    return NSLocalizedString("Email", comment: "Email Notifications Channel")
+                case .Device:
+                    return NSLocalizedString("Push Notifications", comment: "Mobile Push Notifications")
+                }
+            }
+            
+            static let allValues = [ Timeline, Email, Device ]
+        }
+    }
+    
+    // MARK: - Private Properties
+    private let blogPreferenceKeys      = ["new_comment", "comment_like", "post_like", "follow", "achievement", "mentions"]
+    private let blogEmailPreferenceKeys = ["new_comment", "comment_like", "post_like", "follow", "mentions"]
+    private let otherPreferenceKeys     = ["comment_like", "comment_reply"]
+    private let wpcomPreferenceKeys     = ["marketing", "research", "community"]
+    
+    // MARK: - Localized Setting Descriptions
+    private let localizedDescriptionMap = [
+        "new_comment"       : NSLocalizedString("Comments on my site",      comment: "Notifications Settings"),
+        "comment_like"      : NSLocalizedString("Likes on my comments",     comment: "Notifications Settings"),
+        "post_like"         : NSLocalizedString("Likes on my posts",        comment: "Notifications Settings"),
+        "follow"            : NSLocalizedString("Site follows",             comment: "Notifications Settings"),
+        "achievement"       : NSLocalizedString("Site achievements",        comment: "Notifications Settings"),
+        "mentions"          : NSLocalizedString("Username Mentions",        comment: "Notifications Settings"),
+        "comment_reply"     : NSLocalizedString("Replies to your comments", comment: "Notifications Settings"),
+        "marketing"         : NSLocalizedString("Suggestions",              comment: "Notifications Settings"),
+        "research"          : NSLocalizedString("Research",                 comment: "Notifications Settings"),
+        "community"         : NSLocalizedString("Community",                comment: "Notifications Settings")
+    ]
+}
+
+
+
+/**
+*  @brief           NotificationSettings.Channel Equatable Implementation
+*  @details         Swift requires this method to be implemented globally. Sorry about that!
+*
+*  @param           lhs         Left Hand Side Channel
+*  @param           rhs         Right Hand Side Channel
+*  @returns                     A boolean indicating whether two channels are equal. Or not!
+*/
+public func ==(first: NotificationSettings.Channel, second: NotificationSettings.Channel) -> Bool
```

Comment:
I've got my doubts on this one... since it's in the global scope. But... there is no other way to override an operator =(